src/modules/navigator/mission_block.cpp

```diff
@@ -224,7 +224,10 @@ MissionBlock::is_mission_item_reached()
 			 * Therefore the item is marked as reached once the system reaches the loiter
 			 * radius (+ some margin). Time inside and turn count is handled elsewhere.
 			 */
-			if (dist >= 0.0f && dist <= _navigator->get_acceptance_radius(fabsf(_mission_item.loiter_radius) * 1.2f)
+			float radius = (fabsf(_mission_item.loiter_radius) > NAV_EPSILON_POSITION) ? fabsf(_mission_item.loiter_radius) :
```

Comment:
I think this is almost good now. However, I would do `PX4_ISFINITE(_mission_item.loiter_radius)` as the check and not use `fabsf` because that is applied anyway below.

Reply:
I think it's not wrong to have a greater than zero check here. MAV_CMD_NAV_LOITER_TO_ALT has the same kind of issue if the radius is not set and the mavlink specification prescribes "0 means no change to standard loiter".

If you have a mission item with MAV_CMD_NAV_LOITER_TIME with an accidental radius of 0, it is better to use the standard radius than loitering indefinitely.

Reply:
Ah I didn't think of the 0 case!